examples/platforms/nrf528xx/nrf52840/nrf52840.cmake

```diff
@@ -0,0 +1,300 @@
+    PROPERTIES
+        C_STANDARD 99
+        CXX_STANDARD 11
+)
+
+if(OT_EXTERNAL_MBEDTLS)
+    target_link_libraries(openthread-nrf52840
+        PUBLIC
+            ${OT_MBEDTLS}
+            ${NRF52840_3RD_LIBS}
+        PRIVATE
+            openthread-platform-utils
+            ot-config
+    )
+    target_link_libraries(openthread-nrf52840-transport
+        PUBLIC
+            ${OT_MBEDTLS}
+        PRIVATE
+            nordicsemi-nrf52840-sdk
+            ot-config
+    )
+    target_link_libraries(openthread-nrf52840-sdk
+        PUBLIC
+            ${OT_MBEDTLS}
+            ${NRF52840_3RD_LIBS}
+        PRIVATE
+            openthread-platform-utils
+            ot-config
+    )
+    target_link_libraries(openthread-nrf52840-softdevice-sdk
+        PUBLIC
+            ${OT_MBEDTLS}
+            ${NRF52840_3RD_LIBS}
+        PRIVATE
+            openthread-platform-utils
+            ot-config
+    )
+else()
+    target_link_libraries(openthread-nrf52840
+        PUBLIC
+            ${NRF52840_3RD_LIBS}
+        PRIVATE
+            openthread-platform-utils
+            openthread-nrf52840-transport
+            ${OT_MBEDTLS}
+            ot-config
+    )
+    target_link_libraries(openthread-nrf52840-transport
+        PUBLIC
+            nordicsemi-nrf52840-sdk
+        PRIVATE
+            ${OT_MBEDTLS}
+            ot-config
+    )
+    target_link_libraries(openthread-nrf52840-sdk
+        PUBLIC
+            ${NRF52840_3RD_LIBS}
+        PRIVATE
+            openthread-platform-utils
+            ${OT_MBEDTLS}
+            ot-config
+    )
+    target_link_libraries(openthread-nrf52840-softdevice-sdk
+        PUBLIC
+            ${NRF52840_3RD_LIBS}
+        PRIVATE
+            openthread-platform-utils
+            ${OT_MBEDTLS}
+            ot-config
+    )
+endif()
+
+target_link_options(openthread-nrf52840
+    PUBLIC
+        -T${LD_FILE}
+        -Wl,--gc-sections
+        -Wl,-Map=$<TARGET_PROPERTY:NAME>.map
+)
+
+target_link_options(openthread-nrf52840-transport
+    PUBLIC
+        -T${LD_FILE}
+        -Wl,--gc-sections
+        -Wl,-Map=$<TARGET_PROPERTY:NAME>.map
+)
+
+target_link_options(openthread-nrf52840-sdk
+    PUBLIC
+        -T${LD_FILE}
+        -Wl,--gc-sections
+        -Wl,-Map=$<TARGET_PROPERTY:NAME>.map
+)
+
+target_link_options(openthread-nrf52840-softdevice-sdk
+    PUBLIC
+        -T${LD_FILE}
+        -Wl,--gc-sections
+        -Wl,-Map=$<TARGET_PROPERTY:NAME>.map
+)
+
+target_compile_definitions(openthread-nrf52840
+    PUBLIC
+        ${OT_PLATFORM_DEFINES}
+)
+
+target_compile_definitions(openthread-nrf52840-transport
+    PUBLIC
+        ${OT_PLATFORM_DEFINES}
+)
+
+target_compile_definitions(openthread-nrf52840-sdk
+    PUBLIC
+        ${OT_PLATFORM_DEFINES}
+)
+
+target_compile_definitions(openthread-nrf52840-softdevice-sdk
+    PUBLIC
+        ${OT_PLATFORM_DEFINES}
+)
+
+target_compile_options(openthread-nrf52840
+    PRIVATE
+        ${OT_CFLAGS}
+        -DCONFIG_GPIO_AS_PINRESET
```

Comment:
Use a variable for these compile options.
Also move defines to `OT_PLATFORM_DEFINES`.